cpp/arcticdb/processing/clause.hpp

```diff
@@ -511,16 +511,24 @@ struct ColumnStatsGenerationClause {
 struct RowRangeClause {
     enum class RowRangeType: uint8_t {
         HEAD,
-        TAIL
+        TAIL,
+        RANGE
     };
 
     ClauseInfo clause_info_;
     RowRangeType row_range_type_;
     // As passed into head or tail
-    int64_t n_;
+    int64_t n_{0};
 
-    // Row range to keep. Zero-indexed, inclusive of start, exclusive of end
-    // Calculated from n, whether the RowRangeType is head or tail, and the total rows as passed in by set_processing_config
+    // User provided values, which are used to calculate start and end.
+    // Both can be provided with negative values to wrap indices.
+    int64_t user_provided_start_;
+    int64_t user_provided_end_;
+
+    // Row range to keep. Zero-indexed, inclusive of start, exclusive of end.
+    // If the RowRangeType is `HEAD` or `TAIL`, this is calculated from `n` and
+    // the total rows as passed in by `set_processing_config`.
+    // If the RowRangeType is `RANGE`, then start and end are set to the user-provided values.
```

Comment:
I guess this comment is inaccurate if either of the user provided values are negative?

Reply:
Is this clearer?

```suggestion
    // If the RowRangeType is `RANGE`, then start and end are set to the user-provided values
    // in `set_processing_config`.
```